src/server/services/wms/qgswmsrenderer.cpp

```diff
@@ -2411,7 +2417,12 @@ namespace QgsWms
     {
       if ( child->nodeType() == QgsLayerTreeNode::NodeGroup )
       {
-        initLayerGroupsRecursive( static_cast<const QgsLayerTreeGroup *>( child ), child->name() );
+        QString shortName = child->customProperty( QStringLiteral( "wmsShortName" ) ).toString();
+        QString name = child->name();
+        if ( !shortName.isEmpty() )
+          initLayerGroupsRecursive( static_cast<const QgsLayerTreeGroup *>( child ), shortName );
+        else
+          initLayerGroupsRecursive( static_cast<const QgsLayerTreeGroup *>( child ), name );
```

Comment:
It looks good, but I would have preferred something shorter :):

```
QString name = child->customProperty( QStringLiteral( "wmsShortName" ) ).toString();

if ( name.isEmpty() )
  name = child->name();

initLayerGroupsRecursive( static_cast<const QgsLayerTreeGroup *>( child ), name );
```

Reply:
done